scripts/project.js

```diff
@@ -0,0 +1,50 @@
+// Constructor function for portfolio item
+(function(module){
+  function PortfolioItem (opts) {
+    for (key in opts) {
+      this[key] = opts[key];
+    }
+  }
+
+  PortfolioItem.prototype.buildThumbnails = function() {
+    var source = $('#preview-template').html();
+    var template = Handlebars.compile(source);
+    return template(this);
+  };
+
+  PortfolioItem.prototype.updateDetailModal = function(){
+    var source = $('#detail-template').html();
+    var template = Handlebars.compile(source);
+    return template(this);
+  };
+
+  PortfolioItem.loadAll = function(rawData){
+    rawData.sort(function(a,b) {
+      return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+    });
+
+    PortfolioItem.all = rawData.map(function(ele) {
+      return new PortfolioItem(ele);
+    });
+  };
+
+  PortfolioItem.fetchAll = function(){
+    $.getJSON('data/projects.json', function(rawData, status, xhr){
```

Comment:
Refactor: only retrieve header information in this ajax to run the etag check. if you pull from local storage don't run another ajax check, else run another one where you pull the content as well.